pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,53 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"log"
+	"net/http"
+)
+
+func Handler(MaxRequestsInFlight int) http.Handler {
```

Comment:
I am not sure about name of this file. `prometheus.go` doesn't seem so meaningful to me. `handler.go` may be better but I don't have a strong opinion about that.

Reply:
The name could also have something related to "scraper". "metricsScraper" or "prometheusScraper" maybe

Reply:
I don't think so. This part is not related to scraping. This part is related to serving metrics. It is not a big deal. You can ignore this comment if everything else is OK : )